### ISSUE TITLE
Magmatize_data without "Other" reporting group

### ISSUE THREAD

**Issue by csjalbert** [boppingshoe/magma.gsi#2]
@boppingshoe , I'm trying to run MAGMA without an "Other" reporting group and I can't build the magma_data.RData. An easy solution may be to automatically add "Other" to the groupnames file, if not present but I'm not sure this is the 'correct' way to address it.

It looks like the issue is on line 66 of magmatize_data():

`  G <- apply(group_names, 2, function(g) {
    which(g == "Other")
  })`

which leads to failure on line 72:
`groups <- rbind(groups0, matrix(rep(G, length(miss_hatch_grp)), 
    nrow = length(miss_hatch_grp), ncol = length(G), byrow = TRUE, 
    dimnames = list(miss_hatch_grp, names(groups0))))`

Error:
`Error in matrix(rep(G, length(miss_hatch_grp)), nrow = length(miss_hatch_grp),  : 
  length of 'dimnames' [2] not equal to array extent`

Do you think we should just tack on an "Other" group or have any other ideas?

**Comment by csjalbert** [comment]
A bit more here - I added "Other" to `groupnames.txt` and the model worked but I couldn't summarize it with `magmatize_summ()`. There was an error due to NAs or missing values in the dataset. I replaced one of my existing group names with "Other", re-ran `magmatize_data()`, and the summary appears to have worked. I can swap it back on the backend, but I'm not sure we want to modify names like that though, particularly if missing hatcheries or anything else gets added to "Other".

**Comment by boppingshoe** [comment]
I realized what the code was intended to do. When there's an unidentified hatchery group, I programmed magmatize_data() to put it in "Other" group automatically (and list the unidentified groups at the end). It does not stop the data processing. But the code breaks if there's no "Other" in the reporting groups.

I fixed it so now it will still do the same thing as before, but it's ok if you don't have an "Other" group. However, if there's an unidentified hatchery group and no "Other" to put it in, it will stop and display an error.
